### PR TITLE
Top level masonry refactor

### PR DESCRIPTION
**Pull request by FrankFlitton**
# TL;DR

Normalize how parts of pages are related in the DOM to allow for more reliable flex box behavior across screen sizes.

Added a top level layout component that can be substituted in the registry.

- Screen elements resize better with less UI cut off by the window.
- Mobile nav for smaller screen sizes (ie tiling window manager, 1/2 a monitor).
- Standardize flex/flex grid within the app to be more reliable.
- Easier to make and maintain UI components and app pages - composable via MUI's `Grid` framework.

## Type
 - [x] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [ ] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description

Refactored much of the top level code to separate the left nav from the router view. The UI is now flex box based with no top level components having fixed positioning. 

Nav is now programmatically shown and hidden via a context state (with animation). The current approach is hardcoding routes with a wrapper.

During the refactor, it was discovered that the useProjects hook was using a cache system that would dump the payload when the route changed or a new project was selected. The components (left nav and main view) were remounted and created fresh caches on the route changes which masked the issue.

## Tracking Issue
_Remove the '*fixes*' keyword if there will be multiple PRs to fix the linked issue_

fixes https://github.com/flyteorg/flyte/issues/<number>

## Follow-up issue
_NA_

### CODE REVIEW

File: packages/console/src/components/Entities/EntityDetails.tsx [hunk 1]
```diff
@@ -15,6 +16,11 @@ import { EntityVersions } from './EntityVersions';
 import { EntityExecutionsBarChart } from './EntityExecutionsBarChart';
 
 const useStyles = makeStyles((theme: Theme) => ({
+  entityDetailsWrapper: {
+    paddingLeft: theme.spacing(2),
+    paddingRight: theme.spacing(2),
```
Comment (etdotal) [inline packages/console/src/components/Entities/EntityDetails.tsx:21]:
```suggestion
    paddingInline: theme.spacing(2),
```

File: packages/console/src/components/Entities/EntityDetails.tsx [hunk 2]
```diff
@@ -57,58 +63,70 @@ interface EntityDetailsProps {
...
-        {sections.description ? (
-          <div className={styles.descriptionContainer}>
-            <EntityDescription id={id} />
-          </div>
-        ) : null}
-        {!sections.inputs && sections.schedules ? (
-          <div className={styles.schedulesContainer}>
-            <EntitySchedules id={id} />
+          <div className={styles.metadataContainer}>
+            {sections.description ? (
```
Comment (etdotal) [inline packages/console/src/components/Entities/EntityDetails.tsx:81]: can be made more concise

sections.description && (
<div className={styles.descriptionContainer}>
                <EntityDescription id={id} />
              </div>
)

File: packages/console/src/components/Entities/EntityDetails.tsx [hunk 3]
```diff
@@ -57,58 +63,70 @@ interface EntityDetailsProps {
...
+              <div className={styles.descriptionContainer}>
+                <EntityDescription id={id} />
+              </div>
+            ) : null}
+            {!sections.inputs && sections.schedules ? (
```
Comment (etdotal) [inline packages/console/src/components/Entities/EntityDetails.tsx:86]: same here

File: packages/console/src/components/Entities/EntityDetails.tsx [hunk 4]
```diff
@@ -57,58 +63,70 @@ interface EntityDetailsProps {
...
           </div>
-        ) : null}
-      </div>
 
-      {sections.inputs ? (
-        <div className={styles.inputsContainer}>
-          <EntityInputs id={id} />
-        </div>
-      ) : null}
+          {sections.inputs ? (
```
Comment (etdotal) [inline packages/console/src/components/Entities/EntityDetails.tsx:93]: fix formatting (notice the whitespace indicates tabs)

File: packages/console/src/components/Entities/EntityDetails.tsx [hunk 5]
```diff
@@ -57,58 +63,70 @@ interface EntityDetailsProps {
...
+
+          {sections.executions ? (
+            <div className={styles.executionsContainer}>
+              <EntityExecutions
+                chartIds={chartIds}
+                id={id}
+                clearCharts={clearCharts}
+              />
+            </div>
+          ) : (
```
Comment (etdotal) [inline packages/console/src/components/Entities/EntityDetails.tsx:119]: I would avoid use of multiple ? :

If possible, we should define either a new component to handle specific states or define a function `renderExecutions` that does this logic using an if else to make it easy to read and test for

File: packages/console/src/components/Entities/EntityDetails.tsx [hunk 6]
```diff
@@ -57,58 +63,70 @@ interface EntityDetailsProps {
...
+            <>
+              <LoadingSpinner />
+            </>
+          )}
+        </>
+      ) : (
+        <>
```
Comment (etdotal) [inline packages/console/src/components/Entities/EntityDetails.tsx:126]: Loading can be caught earlier and the fragment is unecessary

File: packages/console/src/components/Entities/VersionDetails/EntityVersionDetailsContainer.tsx [hunk 7]
```diff
@@ -40,10 +42,12 @@ const useStyles = makeStyles<Theme, StyleProps>((theme: Theme) => ({
     width: '100%',
     flex: '1',
     overflowY: 'scroll',
+    padding: `0 ${theme.spacing(2)}`,
   },
   versionsContainer: {
     display: 'flex',
     flex: '0 1 auto',
+    padding: `0 ${theme.spacing(2)}`,
```
Comment (etdotal) [inline packages/console/src/components/Entities/VersionDetails/EntityVersionDetailsContainer.tsx:50]:
```suggestion
    padding: theme.spacing(0, 2),
```

File: packages/console/src/components/Entities/VersionDetails/EntityVersionDetailsContainer.tsx [hunk 8]
```diff
@@ -40,10 +42,12 @@ const useStyles = makeStyles<Theme, StyleProps>((theme: Theme) => ({
     width: '100%',
     flex: '1',
     overflowY: 'scroll',
+    padding: `0 ${theme.spacing(2)}`,
```
Comment (etdotal) [inline packages/console/src/components/Entities/VersionDetails/EntityVersionDetailsContainer.tsx:45]:
```suggestion
    padding: theme.spacing(0, 2),
```

File: packages/console/src/components/Entities/VersionDetails/EntityVersionDetailsContainer.tsx [hunk 9]
```diff
@@ -26,6 +27,7 @@ const useStyles = makeStyles<Theme, StyleProps>((theme: Theme) => ({
     flexWrap: 'nowrap',
     overflow: 'hidden',
     height: `calc(100vh - ${theme.spacing(17)}px)`,
+    padding: `0 ${theme.spacing(2)}`,
```
Comment (etdotal) [inline packages/console/src/components/Entities/VersionDetails/EntityVersionDetailsContainer.tsx:30]:
```suggestion
    padding: theme.spacing(0, 2),
```

File: packages/console/src/components/Entities/VersionDetails/EntityVersionDetailsContainer.tsx [hunk 10]
```diff
@@ -82,33 +86,43 @@ const EntityVersionsDetailsContainerImpl: React.FC<
...
-            <StaticGraphContainer workflowId={workflowId} />
-          </div>
-        )}
-        <div className={styles.versionsContainer}>
-          <EntityVersions id={id} showAll />
-        </div>
-      </div>
-    </WaitForData>
+        </>
+      ) : (
```
Comment (etdotal) [inline packages/console/src/components/Entities/VersionDetails/EntityVersionDetailsContainer.tsx:120]: Please catch the loader state early via a return

if (!project.id) {
  return <LoadingSpinner />
}


...

File: packages/console/src/components/Executions/ExecutionDetails/ExecutionMetadata.tsx [hunk 11]
```diff
@@ -18,19 +18,15 @@ const useStyles = makeStyles((theme: Theme) => {
...
-      flexDirection: 'column',
-      position: 'relative',
+      width: '100%',
     },
     detailsContainer: {
-      alignItems: 'start',
       display: 'flex',
-      flex: '0 1 auto',
-      paddingTop: theme.spacing(3),
+      paddingTop: theme.spacing(1),
```
Comment (etdotal) [inline packages/console/src/components/Executions/ExecutionDetails/ExecutionMetadata.tsx:25]: use `paddingBlockStart` and `paddingBlockEnd` instead of top and bottom

File: packages/console/src/components/Executions/ExecutionDetails/Timeline/TimelineChart/utils.ts [hunk 12]
```diff
@@ -74,7 +74,7 @@ export const getOperationsFromWorkflowExecutionMetrics = (
   const operationIds = uniq(
     traverse(data)
       .paths()
-      .filter(path => path.at(-1) === 'operationId')
+      .filter(path => path[path.length - 1] === 'operationId')
```
Comment (etdotal) [inline packages/console/src/components/Executions/ExecutionDetails/Timeline/TimelineChart/utils.ts:77]: this should work with .at(-1). Was typescript complaining?

File: packages/console/src/components/Executions/ExecutionFilters.tsx [hunk 13]
```diff
@@ -18,9 +18,10 @@ const useStyles = makeStyles((theme: Theme) => ({
     alignItems: 'center',
     display: 'flex',
     flexDirection: 'row',
-    height: theme.spacing(7),
     minHeight: theme.spacing(7),
     paddingLeft: theme.spacing(1),
+    paddingTop: theme.spacing(2),
```
Comment (etdotal) [inline packages/console/src/components/Executions/ExecutionFilters.tsx:23]: use paddingBlockStart and paddingBlockEnd

File: packages/console/src/components/Navigation/DefaultAppBarContent.tsx [hunk 14]
```diff
@@ -33,13 +28,53 @@ interface DefaultAppBarProps {
...
   const { adminVersion } = useAdminVersion();
   const isGAEnabled = env.ENABLE_GA === 'true' && env.GA_TRACKING_ID !== '';
 
+  const handleSideNavToggle = React.useCallback(() => {
+    if (isSideNavOpen) {
+      closeSideNav();
+    } else {
+      openSideNav();
+    }
+  }, [isSideNavOpen, openSideNav, closeSideNav]);
```
Comment (etdotal) [inline packages/console/src/components/Navigation/DefaultAppBarContent.tsx:53]: use an early return

if (isSideNavOpen) {
  closeCloseSideNav();
  return;
}

openSideNav();

File: packages/console/src/components/Navigation/DefaultAppBarContent.tsx [hunk 15]
```diff
@@ -33,13 +28,53 @@ interface DefaultAppBarProps {
...
+
+  const theme = useTheme();
+
+  // Enable / Disable mobile nav behaviour based on screen size
+  React.useLayoutEffect(() => {
+    const handleResize = () => {
+      if (window.innerWidth < theme.breakpoints.values.md) {
+        if (!isMobileNav) {
```
Comment (etdotal) [inline packages/console/src/components/Navigation/DefaultAppBarContent.tsx:61]: There is an unnecessary if statement inside another if. Please group the condition using &&.

You should also return early if an if statement is not satisfied since it prevents code nesting and easy to read

File: packages/console/src/components/Navigation/DefaultAppBarContent.tsx [hunk 16]
```diff
@@ -74,27 +109,103 @@ export const DefaultAppBarContent = (props: DefaultAppBarProps) => {
     },
   ];
 
+  const styles = makeStyles(() => ({
+    wordmark: {
+      position: 'relative',
+      paddingTop: '22px',
```
Comment (etdotal) [inline packages/console/src/components/Navigation/DefaultAppBarContent.tsx:115]: use theme values for padding. This can be theme.spacing(2.75)

File: packages/console/src/components/Navigation/DefaultAppBarContent.tsx [hunk 17]
```diff
@@ -74,27 +109,103 @@ export const DefaultAppBarContent = (props: DefaultAppBarProps) => {
...
+              to={Routes.SelectProject.path}
+            >
+              <FlyteLogo size={32} hideText={isLayoutHorizontal} />
+              {isLayoutHorizontal && (
+                <Box className={styles.wordmark}>
+                  <FlyteLogo size={32} />
+                </Box>
+              )}
+            </Link>
+            {props.items?.length > 0 ? (
```
Comment (etdotal) [inline packages/console/src/components/Navigation/DefaultAppBarContent.tsx:174]: We should return null and not false. Also this can be made shorter using && instead of ?:

File: packages/console/src/components/Navigation/NavBar.tsx [hunk 18]
```diff
@@ -17,6 +20,32 @@ const DefaultAppBarContent = lazy(() => import('./DefaultAppBarContent'));
...
+      top: 0,
+    },
+    inlineNavBar: {
+      width: '80px',
+      height: '100%',
+      position: 'fixed',
+      inset: '0',
+    },
+    inlineToolBar: {
+      padding: `${theme.spacing(2)}px 0 ${theme.spacing(4)}px 0`,
```
Comment (etdotal) [inline packages/console/src/components/Navigation/NavBar.tsx:42]:
```suggestion
      padding: theme.spacing(2, 0),
```

**Review by etdotal — DISMISSED** [review]
left initial set of comments

File: packages/console/src/components/Entities/EntityDetails.tsx (hunk 1 above)
Comment (etdotal) [reply packages/console/src/components/Entities/EntityDetails.tsx:21]: I personally think paddingInline does the same stuff as paddingLeft/right but additionally also takes care of localization. For a RTL language, you might want padding on the right side rather than the left or something similar and since we support internationalization, we should make sure to use apt spacing attributes/values.

Confirmed that padding inline and block work with MUI as well so we should try to use it when possible unless we are explicitly sure that we want to use padding left which didnt seem to be the case in here.

**Review by jsonporter — APPROVED** [review]

**Review by ursucarina — APPROVED** [review]

**Review by jsonporter — APPROVED** [review]

**Review by ursucarina — APPROVED** [review]

**Review by ursucarina — COMMENTED** [review]
You might need to up the package versions for components, console and update website as well

**Review by jsonporter — DISMISSED** [review]

**Review by jsonporter — APPROVED** [review]
